### PR TITLE
exposed request permissions methods for ios

### PR DESCRIPTION
**Pull request by aerlinn13**
This PR adds requestMicPermission and requestCameraPermission methods to invoke in mobile app. Please note that completion handler for the requestCameraPermission is invoked on main thread to prevent twilio UI to be opened.

### CODE REVIEW

**Review by aleksey-pyatlin — APPROVED** [review]

File: src/js/CordovaCall.js [hunk 1]
```diff
@@ -123,3 +123,11 @@ exports.callNumber = function(to, success, error) {
 exports.reportCallEndedReason = function (callId, reason, success, error) {
     exec(success, error, "CordovaCall", "reportCallEndedReason", [callId, reason]);
 }
+
+exports.requestMicPermission = function(success, error) {
```
Comment (FreddyMcRay) [inline src/js/CordovaCall.js:127]: keeping in mind the fact that results of those operations will be used for some further processing inside cordova app I'd suggest to wrap them into Promises.

**Review by FreddyMcRay — COMMENTED** [review]
Actually I thought that we will use the api provided in twilio plugin (the work Brian just recently did). What's the matter to duplicate "permissions stuff" here as well.

**Comment by aerlinn13** [comment]
> Actually I thought that we will use the api provided in twilio plugin (the work Brian just recently did). What's the matter to duplicate "permissions stuff" here as well.

I am actually fan of reusing stuff, but only unless I have a reason not to do that, for following reasons.

1. First, my implementation of requestCameraPermission blocks main thread to prevent further UI updates (opening actual twilio UI, which it currently does, and we need to avoid it). Reusing it for twilio video UI can cause UI break, so it could take time to make it work everywhere.
2. There is just two small methods which use single one AVFoundation method, and it's all in one, it requests permissions if not granted and passes back status if not determined, cordova part does not really need these getters, native part should handle that without making cordova concerned.
3. From phone handler perspective, all logic of call preparation and orchestration is handled by callkit, whereas twilio plugin is responsible of twilio UI. I believe permissions belong to call orchestration as it's crucial to obtain them before starting the call to avoid missing audio / video.

> keeping in mind the fact that results of those operations will be used for some further processing inside cordova app I'd suggest to wrap them into Promises.

I am not sure I understood. They already have callbacks for success and error which are executed by cordova, why would we need promises?

@LorinWeb will appreciate your opinion on that given we have very tight timeframe to deliver that.

**Comment by FreddyMcRay** [comment]
> I am not sure I understood. They already have callbacks for success and error which are executed by cordova, why would we need promises?

I meant that calling this way
await cordovaCall.requestMicPermission()

is more compact and readable than this
cordovaCall.requestMicPermission(() => console.log('granted'), () => console.log('rejected'))

File: src/js/CordovaCall.js (hunk 1 above)
Comment (cchatdev) [reply src/js/CordovaCall.js:127]: I agree, can we please add that to a separate task? So that it's not lost

**Review by clearbrian — APPROVED** [review]
looks ok

**Pull request merged**